theme/static/js/hs_file_browser.js

```diff
@@ -1289,6 +1367,85 @@ function refreshFileBrowser(name) {
+                var cookie = jQuery.trim(cookies[i]);
+                // Does this cookie string begin with the name we want?
+                if (cookie.substring(0, name.length + 1) == (name + '=')) {
+                    cookieValue = decodeURIComponent(cookie.substring(name.length + 1));
+                    break;
+                }
+            }
+        }
+        return cookieValue;
+    }
+    function csrfSafeMethod(method) {
+        // these HTTP methods do not require CSRF protection
+        return (/^(GET|HEAD|OPTIONS|TRACE)$/.test(method));
+    }
+    function sameOrigin(url) {
+        // test that a given url is a same-origin URL
+        // url could be relative or scheme relative or absolute
+        var host = document.location.host; // host + port
+        var protocol = document.location.protocol;
+        var sr_origin = '//' + host;
+        var origin = protocol + sr_origin;
+        // Allow absolute or scheme relative URLs to same origin
+        return (url == origin || url.slice(0, origin.length + 1) == origin + '/') ||
+            (url == sr_origin || url.slice(0, sr_origin.length + 1) == sr_origin + '/') ||
+            // or any other URL that isn't scheme relative or absolute i.e relative.
+            !(/^(\/\/|http:|https:).*/.test(url));
+    }
+    $.ajaxSetup({
+        beforeSend: function(xhr, settings) {
+            if (!csrfSafeMethod(settings.type) && sameOrigin(settings.url)) {
+                // Send the token to same-origin, relative URLs only.
+                // Send the token only if the method warrants CSRF protection
+                // Using the CSRFToken value acquired earlier
+                xhr.setRequestHeader("X-CSRFToken", getCookie('csrftoken'));
+            }
+        }
+    });
+    $('#btn-confirm-delete').prop('disabled', true)
+    $('#additional-citation-warning').text('Analyzing files . . .')
+    $.ajax({
+        type: "GET",
+        url: '/hsapi/resource/' + shortId + '/file_list/',
+    }).complete(function(res) {
+        if (res.responseText) {
+            let pathObjs = JSON.parse(res.responseText).results
+            try {
+                var external_links = Number.parseInt(ext_link_count);
+            }
+            catch {
+                var external_links = 0;
+            }
+            if (custom_citation !== 'None' && external_links > 0 && countSelectedExternalFiles(pathObjs) >= external_links) {
+                removeCitationIntent = true;
+                $('#additional-citation-warning').text('Removing all referenced content from this resource will also ' +
+                  'remove the custom citation you have entered. Are you sure you want to remove this reference content ' +
+                  'and custom citation?')
+            } else {
+                removeCitationIntent = false;
+                $('#additional-citation-warning').text('')
+            }
+        }
+        $('#btn-confirm-delete').prop('disabled', false)
+    })
+}
+
```

Comment:
I don't understand why you need to set up csrf inside JS function. Is this function triggered from a html element in a form? If so, you could leverage Django's csrf_token tag, right?

Reply:
I haven't finished documenting the monolithic structure of Resource Landing Page, but the single, massive scope for variables and assuming that everything will be available (somewhere) should start to be properly dismantled. I am pretty sure this code reads the existing cookie and reuses it. Depending on where items are imported via the Django directives and when things are rendered, it may not be the best idea to just assume CSRF is handled, especially since file manager is on it's own in a pure JavaScript file. I think most of the POSTs for forms are accompanied with the CSRF Django tag, which isn't available inside this JS file. Also, there weren't any POSTS in this file/no RESTful action, so I was starting a healthy pattern of looking for a cookie. This will generalize well, into the future, espeically as the global scope issue is cleaned up.

Reply:
I can remove it for now and instead leave a comment.

Reply:
Anyway, the theme here is more cohesive, discretized units of script code, html and styling, instead of one massive chunk of 15,000 or however many lines of code. I'm still pulling together the numbers.